src/hotspot/cpu/aarch64/templateInterpreterGenerator_aarch64.cpp

```diff
@@ -811,7 +811,8 @@ void TemplateInterpreterGenerator::lock_method() {
   __ check_extended_sp();
   __ sub(sp, sp, entry_size); // add space for a monitor entry
   __ sub(esp, esp, entry_size);
-  __ mov(rscratch1, sp);
+  __ sub(rscratch1, sp, rfp);
+  __ asr(rscratch1, rscratch1, Interpreter::logStackElementSize);
```

Comment:
> The shifts when calculating the relativized extended SP increase latency and reduce throughput, and they don't do anything useful.

It's true that the shift instructions add a bit of unnecessary overhead, but all the relativize and derelativize code today is based on pointer arithmetic. For instance interpreter frame locals are found at:

   `fp() + *addr_at(interpreter_frame_locals_offset)`

So in order to get rid of the shifts we would need to move away from pointer arithmetic, which means adding a lot of explicit casts in the C++ code. To get the pointer to the locals the code above would have to be changed into:

   `(intptr_t*)((char *)fp() + (char *)*addr_at(interpreter_frame_locals_offset))`

I choose to stick to the definition where a relativized value is "a word index from the frame pointer" and not change it to "number of bytes from the frame pointer", because it meant less changes to the C++ code.

Some interpreter frame members (such as `extended_sp`) doesn't show up in the C++ code as much as `locals`, but I didn't want to have different representations for a relativized pointer, so all of them are "word index from the frame pointer".

Also, the overhead of the shifts is likely negligibly since we now calculate the relativized value once, and not every time we freeze and thaw.

Reply:
OK, fair enough. It's somewhat annoying and pointless, but we can live with it.